### DISCUSSION TITLE
How to get the surface normal of another object

### DISCUSSION THREAD

**Discussion by HSDSZ** [bnpr/Malt#539] [Q&A] [answered]
In eevee, this could be done via geometry node (saving normal in geometry attribute then get it back shader node)
In malt, is there any workaround?

**Comment by HSDSZ** [comment]
Actually, what I want is a stylized shaded tree.

**Comment by HSDSZ** [comment]
Actually, what i want is a stylized shaded tree (left), but useing the right geometry normal for the NPR diffuse normal socket
![image](https://github.com/bnpr/Malt/assets/36077492/074b7fbd-dc07-466e-99b2-61b35dd4b81a)
![image](https://github.com/bnpr/Malt/assets/36077492/93edbf39-631d-4e14-beb2-2965aba782c5)

With anoter geometry normal, the tree will get a general diffuse gradient instead of each leaf having a separted one

**Comment by HSDSZ** [comment]
Here is a demo using the normal of another geometry
![image](https://github.com/bnpr/Malt/assets/36077492/f3d05354-9c5c-48e2-95ba-a7e4174faab7)

**Comment by pragma37** [comment]
You can use geometry nodes as well with vertex colors.
Just keep in mind that only Face Corner Color Attributes are supported.

**Reply by HSDSZ** [reply]
Is this a bug? malt vertex color is not working

in eevee, vertex color working correctly
![image](https://github.com/bnpr/Malt/assets/36077492/c2e67d4a-96f8-4521-80b9-e90284f78b9f)

in malt
![image](https://github.com/bnpr/Malt/assets/36077492/7405bd70-6c87-4fcc-ab54-aaf344db975f)

here is the blend file
[vertex color.zip](https://github.com/bnpr/Malt/files/13727480/vertex.color.zip)

**Reply by pragma37** [reply]
It looks like a bug (this used to work fine), but in Blender itself.
You can check that applying the geometry nodes modifier deletes the Color Attribute.
It looks like the same is happening with the mesh that Malt receives from Blender.

**Reply by pragma37** [reply, marked as answer]
Ok, it's not a bug. You have to ensure that your output is set to *Face Corner > Color*:
![imagen](https://github.com/bnpr/Malt/assets/36610999/60373f74-c95b-4547-b009-5ce30ea8c3f1)

**Reply by HSDSZ** [reply]
oh, i see. I also need the change the face corner color in geometry node not only in the right panel
